source/module.c

```diff
@@ -931,6 +932,37 @@ static void s_handle_failed_callback(napi_env env, napi_value function, napi_sta
     }
 }
 
+napi_status aws_napi_create_external_arraybuffer_function(
+    napi_env env,
+    void *external_data,
+    size_t byte_length,
+    napi_finalize finalize_cb,
+    void *finalize_hint,
+    napi_value *result) {
+
+    napi_status external_buffer_status =
+        napi_create_external_arraybuffer(env, external_data, byte_length, finalize_cb, finalize_hint, result);
+
+    if (external_buffer_status != napi_ok) {
+
+        /* TODO: The enum `napi_no_external_buffers_allowed` is introduced in node14.
+         * Use it to determine if the function failed because of the external buffer support after bump
+         * minimal support to node 14
+         */
+        // The external buffer is disabled, manually copy the external_data into Node
+        void *napi_buf_data = NULL;
+        AWS_NAPI_ENSURE(env, napi_create_arraybuffer(env, byte_length, &napi_buf_data, result));
+
+        memcpy(napi_buf_data, external_data, byte_length);
+
+        // As the data has been copied into the Node, invoke the finalize callback to make sure the
+        // data is released.
+        finalize_cb(env, finalize_hint, finalize_hint);
+    }
+
+    return external_buffer_status;
```

Comment:
if we take the if-branch, and then succeed, `external_buffer_status` will still have an error value

Reply:
Updated the `aws_napi_create_external_arraybuffer`. Now it would return the error code from `napi_create_arraybuffer` instead of assert the function when failed. It will also return `napi_ok` for a successful path. The caller would be responsible to handle the `exteranl_buffer` when the function failed.